lab_4_summarization_textrank/main.py

```diff
@@ -2,77 +2,123 @@
 Lab 4
 Summarize text using TextRank algorithm
 """
-from typing import Union
+import re
+from typing import Union, Any, Type
 
 from lab_3_keywords_textrank.main import TextEncoder, \
-    TextPreprocessor
+    TextPreprocessor, TFIDFAdapter
 
 PreprocessedSentence = tuple[str, ...]
 EncodedSentence = tuple[int, ...]
 
 
+def check_type(user_var: Any, expected_type: Union[Type, tuple[Type, ...]]) -> None:
+    """
+    Checks whether type of user_var is expected_type,
+    if not - raises ValueError
+    """
+    if not isinstance(user_var, expected_type):
+        raise ValueError
+    if expected_type == int and isinstance(user_var, int) and isinstance(user_var, bool):
+        raise ValueError
+
+
+def check_collection(user_var: Any,
+                     expected_elements_type: Type,
+                     *expected_collection_type: Type,
+                     can_be_empty: bool = False) -> None:
+    """
+    Checks whether type of user_var is at least one of expected_collection_type,
+    checks whether type of elements in user_var are expected_elements_type,
+    if not - raises ValueError
+    if expected_elements_type is None, doesn't check elements
+    """
+    num_errors = 0
+    if can_be_empty is False and not user_var:
+        raise ValueError
+    for i in expected_collection_type:
+        try:
+            check_type(user_var, i)
+        except ValueError:
+            num_errors += 1
+    if num_errors == len(expected_collection_type):
```

Comment:
If checks failed not for all elements, that it is ok?

Reply:
fixed